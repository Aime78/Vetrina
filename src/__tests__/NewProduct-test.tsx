import React from 'react';
import {
  AllProductsButton,
  GeneralInfo,
  Price,
  ProductType,
} from 'screens/NewProductScreen/components';
import renderer from 'react-test-renderer';

it('renders the all products button of the new product screen appropriately', () => {
  const tree = renderer.create(<AllProductsButton />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the general information of the product screen appropriately', () => {
  const tree = renderer.create(<GeneralInfo />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the price information of the product screen appropriately', () => {
  const tree = renderer.create(<Price />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the product type of the product screen appropriately', () => {
  const tree = renderer.create(<ProductType />).toJSON();
  expect(tree).toMatchSnapshot();
});
