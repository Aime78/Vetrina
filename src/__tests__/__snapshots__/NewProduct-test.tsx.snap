// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the all products button of the new product screen appropriately 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": "#21B8F9",
        "borderRadius": 105,
        "borderWidth": 1,
        "flexDirection": "row",
        "gap": 15,
        "height": 50,
        "justifyContent": "center",
        "width": 147,
      },
      {
        "shadowColor": "#171717",
        "shadowOffset": {
          "height": 4,
          "width": -2,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 3,
      },
    ]
  }
>
  <Image
    source={
      {
        "testUri": "../../../src/assets/chevron-left-blue.png",
      }
    }
  />
  <Text
    style={
      {
        "color": "#103B66",
        "fontSize": 16,
        "fontWeight": "600",
      }
    }
  >
    All products
  </Text>
</View>
`;

exports[`renders the general information of the product screen appropriately 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": 460,
        "marginTop": 30,
        "padding": 20,
        "width": 370,
      },
      {
        "shadowColor": "#171717",
        "shadowOffset": {
          "height": 4,
          "width": -2,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      },
    ]
  }
>
  <Text
    style={
      {
        "color": "#103B66",
        "fontSize": 22,
        "fontWeight": "600",
        "marginBottom": 30,
      }
    }
  >
    General Information
  </Text>
  <View>
    <Text
      style={
        {
          "color": "#546679",
          "fontSize": 15,
          "fontWeight": "600",
        }
      }
    >
      Product name
    </Text>
    <View
      style={
        {
          "borderColor": "1 solid rgba(10, 37, 64, 0.32)",
          "borderRadius": 5,
          "borderWidth": 1,
          "marginBottom": 30,
          "marginTop": 10,
        }
      }
    >
      <TextInput
        placeholder="Lorem ipsum"
        style={
          {
            "color": "#8592A0",
            "fontSize": 18,
            "fontWeight": "400",
            "height": 40,
            "padding": 10,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "position": "relative",
      }
    }
  >
    <Text
      style={
        {
          "color": "#546679",
          "fontSize": 15,
          "fontWeight": "600",
        }
      }
    >
      Description
    </Text>
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "borderBottomColor": "1 solid rgba(10, 37, 64, 0.32)",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "gap": 30,
            "left": 0,
            "padding": 17,
            "position": "absolute",
            "top": 10,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../src/assets/format-bold.png",
            }
          }
        />
        <Image
          source={
            {
              "testUri": "../../../src/assets/italic.png",
            }
          }
        />
        <Image
          source={
            {
              "testUri": "../../../src/assets/format-t.png",
            }
          }
        />
        <Image
          source={
            {
              "testUri": "../../../src/assets/format-underlined.png",
            }
          }
        />
        <Image
          source={
            {
              "testUri": "../../../src/assets/format-list.png",
            }
          }
        />
        <Image
          source={
            {
              "testUri": "../../../src/assets/format-link.png",
            }
          }
        />
      </View>
      <View
        style={
          {
            "borderColor": "1 solid rgba(10, 37, 64, 0.32)",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginBottom": 30,
            "marginTop": 10,
          }
        }
      >
        <TextInput
          placeholder="Description (0 / 5000)"
          style={
            {
              "color": "#8592A0",
              "fontSize": 18,
              "fontWeight": "400",
              "height": 230,
              "padding": 10,
              "paddingTop": 1,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`renders the price information of the product screen appropriately 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": 320,
        "marginTop": 30,
        "padding": 20,
        "width": 370,
      },
      {
        "shadowColor": "#171717",
        "shadowOffset": {
          "height": 4,
          "width": -2,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      },
    ]
  }
>
  <Text
    style={
      {
        "color": "#103B66",
        "fontSize": 22,
        "fontWeight": "600",
        "marginBottom": 30,
      }
    }
  >
    Price
  </Text>
  <View
    style={
      {
        "position": "relative",
      }
    }
  >
    <Text
      style={
        {
          "color": "#546679",
          "fontSize": 15,
          "fontWeight": "600",
        }
      }
    >
      Original price
    </Text>
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 10,
            "left": 10,
            "position": "absolute",
            "top": 22,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../src/assets/euro-input.png",
            }
          }
        />
        <Text
          style={
            {
              "color": "#103B66",
              "fontSize": 14,
              "fontWeight": "400",
            }
          }
        >
          0
        </Text>
      </View>
      <View
        style={
          {
            "borderColor": "1 solid rgba(10, 37, 64, 0.32)",
            "borderRadius": 5,
            "borderWidth": 1,
            "marginBottom": 30,
            "marginTop": 10,
          }
        }
      >
        <TextInput
          style={
            {
              "color": "#8592A0",
              "fontSize": 18,
              "fontWeight": "400",
              "height": 40,
              "paddingLeft": 38,
            }
          }
        />
      </View>
    </View>
  </View>
  <View>
    <Text
      style={
        {
          "color": "#546679",
          "fontSize": 15,
          "fontWeight": "600",
        }
      }
    >
      Discounted price
    </Text>
    <View
      style={
        {
          "position": "relative",
        }
      }
    >
      <View
        style={
          {
            "left": 10,
            "position": "absolute",
            "top": 20,
            "zIndex": 1,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../src/assets/euro-icon.png",
            }
          }
        />
      </View>
      <View
        style={
          {
            "backgroundColor": "#E7E9EC",
            "borderRadius": 5,
            "marginBottom": 30,
            "marginTop": 10,
          }
        }
      >
        <TextInput
          placeholder="Lorem ipsum"
          style={
            {
              "fontSize": 14,
              "height": 40,
              "padding": 10,
              "paddingLeft": 40,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 10,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/checkbox.png",
        }
      }
    />
    <Text
      style={
        {
          "color": "#103B66",
          "fontSize": 14,
          "fontWeight": "400",
        }
      }
    >
      Activate discounted price
    </Text>
  </View>
</View>
`;

exports[`renders the product type of the product screen appropriately 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "white",
        "borderRadius": 10,
        "height": 150,
        "marginBottom": 60,
        "marginTop": 30,
        "padding": 20,
        "width": 370,
      },
      {
        "shadowColor": "#171717",
        "shadowOffset": {
          "height": 4,
          "width": -2,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 3,
      },
    ]
  }
>
  <Text
    style={
      {
        "color": "#103B66",
        "fontSize": 22,
        "fontWeight": "600",
        "marginBottom": 30,
        "marginTop": 20,
      }
    }
  >
    Product type
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "gap": 20,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 10,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "borderColor": "#8592A0",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "#21B8F9",
              "borderRadius": 6,
              "height": 12,
              "width": 12,
            }
          }
        />
      </View>
      <Text
        style={
          {
            "color": "#103B66",
            "fontSize": 17,
            "fontWeight": "400",
          }
        }
      >
        Physical
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 10,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "borderColor": "#8592A0",
            "borderRadius": 12,
            "borderWidth": 2,
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          }
        }
      />
      <Text
        style={
          {
            "color": "#103B66",
            "fontSize": 17,
            "fontWeight": "400",
          }
        }
      >
        Digital
      </Text>
    </View>
  </View>
</View>
`;
